shim.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/moby/moby/pkg/term"
+)
+
+type Shim struct {
+	container string
+	process   string
```

Comment:
- Presumably `container` is the `containerID`? If so, I think it would be clearer to name it with the `ID` suffix.
- Is `process` the PID? If so, again, why not specify it as `pid int`?

Reply:
processes'  numeric pid are not seen on the host, we don't want to expose them, right?
{containerID, processID} is used to refer the init/exec process in the container.
Both containerID and processID are strings.

(Linux processes spawned by init/exec process which are supposed to be invisible in the host can't not be referenced.)